events/views.py

```diff
@@ -117,74 +129,160 @@ def __init__(self, **kwargs) -> None:
 
         is_valid_formsets = False
-        if self.create_event_form.is_valid():
+        if self.create_event_form.is_valid() and self.reminder_form.is_valid():
             event_instance = self.create_event_form.save()
             event_creator = EventParticipant.objects.get(event_id=event_instance, user_id=request.user, is_creator=True)
 
             if self.check_optional_meeting_dates_formset(
-               event_instance, event_creator, self.optional_meetings_formset):
-                if self.check_participant_formset(request, event_instance, self.meeting_participants_formset):
+               event_instance, event_creator, self.optional_meetings_formset, meeting_instance):
+                if self.check_participant_formset(
+                    request, event_instance, self.meeting_participants_formset, meeting_instance
+                ):
                     # all the forms are valid and all the data saved in the DB
                     is_valid_formsets = True
 
-                    timeout = EventPlanner.get_timeout(event_instance)
-                    EventPlanner.creating_meeting_reminders(event_creator, timeout)
-                    for message in EventPlanner.get_timeout_message(timeout).split("\n"):
-                        messages.success(request, message)
-                    EventPlanner.send_invite_notification(event_instance)
-                    EventPlanner.send_meeting_invitation_email_to_participants(event_instance, event_creator)
+                    reminder = self.reminder_form.save(commit=False)
+
+                    if reminder.date_time:
+                        reminder.participant_id = event_creator
+                        reminder.messages = convert_time_delta(event_instance.date_time_start - reminder.date_time)
+                        reminder.save()
+
+                    CreateMeetingView.calc_timeout_and_send_meeting_notifications(
+                        request, event_instance, event_creator, meeting_instance, self.min_date_change
+                    )
                     return redirect(HOME_PAGE)
             if not is_valid_formsets:
                 # getting all the data that the user entered in the forms
                 self.formset_meeting_data = self.get_formset_meeting_date(self.optional_meetings_formset)
                 self.formset_participant_data = self.get_formset_participant_date(request)
         return self.render_to_response(self.get_context_data())
 
+    @staticmethod
+    def calc_timeout_and_send_meeting_notifications(request, event_instance, event_creator,
+                                                    updated_meeting, min_date_change):
+        timeout = EventPlanner.get_timeout(event_instance)
+
+        if updated_meeting and min_date_change:
+            Reminder.objects.remove_algorithm_reminder(event_creator)
+            EventPlanner.creating_meeting_reminders(event_creator, timeout)
+
+        meeting_status = "created"
+        if updated_meeting:
```

Comment:
please move it to one line 
```suggestion
         meeting_status = "created" if updated_meeting else "updated"
```

Reply:
There is more logic that depends on the `if updated_meeting` condition besides changing the value of `meeting_status`.